### PR TITLE
Multiselection in filter

### PR DESCRIPTION
**Pull request by rdubigny**

### CODE REVIEW

**Comment by rdubigny** [comment]
[ETQ instructeur, dans les filtres par type d'api dans ma vue liste, je ne vois que les API dont je suis administrateur.](https://trello.com/c/MQk2rCps/873-etq-instructeur-dans-les-filtres-par-type-dapi-dans-ma-vue-liste-je-ne-vois-que-les-api-dont-je-suis-administrateur)

File: src/lib/index.js [hunk 1]
```diff
@@ -246,6 +246,10 @@ export const getStateFromUrlParams = (defaultState = {}) => {
           return { id: k, desc: v === 'desc' };
```
Comment (XavierJp) [inline src/lib/index.js:246]: I am a bit lost. Ithought we kept `getStateFromUrlParams`for legacy purposes ? If it is the case we shouldnot need to implement `,` as past urlparams should not use it ?

File: src/pages/AdminEnrollmentList.js [hunk 2]
```diff
@@ -142,19 +151,49 @@ class AdminEnrollmentList extends React.Component {
...
+
+        const value = filter
+          ? isArray(filter.value)
+            ? filter.value
+            : [filter.value]
+          : [''];
+
+        return (
+          <select
+            onChange={event => {
```
Comment (XavierJp) [inline src/pages/AdminEnrollmentList.js:200]: maybe declare the onchange function before for readibility

File: src/pages/AdminEnrollmentList.js [hunk 3]
```diff
@@ -142,19 +151,49 @@ class AdminEnrollmentList extends React.Component {
...
+          .map(role => {
+            const targetApiKey = role.split(':')[0];
+
+            return {
+              key: targetApiKey,
+              label: TARGET_API_LABELS[targetApiKey],
+            };
+          });
+
+        const value = filter
```
Comment (XavierJp) [inline src/pages/AdminEnrollmentList.js:192]: maybe a more self explanatory variable here like `selectedvalue` or ìnitialvalue`

File: src/pages/AdminEnrollmentList.js [hunk 4]
```diff
@@ -142,19 +151,49 @@ class AdminEnrollmentList extends React.Component {
...
+          ? isArray(filter.value)
+            ? filter.value
+            : [filter.value]
```
Comment (XavierJp) [inline src/pages/AdminEnrollmentList.js:195]: double ternary is very hard to read. I wonder what are the input and output signature now

File: src/pages/AdminEnrollmentList.js [hunk 5]
```diff
@@ -188,19 +227,37 @@ class AdminEnrollmentList extends React.Component {
...
-          <option value="">Tous</option>
-          {toPairs(ADMIN_STATUS_LABELS).map(([key, label]) => (
-            <option key={key} value={key}>
-              {label}
-            </option>
-          ))}
-        </select>
-      ),
+      Filter: ({ filter, onChange }) => {
+        const value = filter
```
Comment (XavierJp) [inline src/pages/AdminEnrollmentList.js:261]: it looks a lot like the previous ternary, maybe create an helper function you can document that will be easier to understand what it is doing

File: src/pages/AdminEnrollmentList.js [hunk 6]
```diff
@@ -188,19 +227,37 @@ class AdminEnrollmentList extends React.Component {
...
+          ? isArray(filter.value)
+            ? filter.value
+            : [filter.value]
+          : [''];
+
+        return (
+          <select
+            onChange={event => {
+              // get multiple options from react select
```
Comment (XavierJp) [inline src/pages/AdminEnrollmentList.js:270]: same comment

File: src/pages/AdminEnrollmentList.js [hunk 7]
```diff
@@ -188,19 +227,37 @@ class AdminEnrollmentList extends React.Component {
...
+              // see https://stackoverflow.com/questions/28624763/retrieving-value-from-select-with-multiple-option-in-react
```
Comment (XavierJp) [inline src/pages/AdminEnrollmentList.js:271]: nice to reference stack overflow :)

**Review by XavierJp — COMMENTED** [review]
I spoted a few naming & refacto improvements 

I still need to test it though

**Pull request merged**